controllers/secret_controller.go

```diff
@@ -64,9 +65,9 @@ func (r *SecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		r.log.Error(err, "Unable to retrieve private key, please ensure it is created")
 	}
 
-	privateKeyPredicate := builder.WithPredicates(predicate.Funcs{
+	secretPredicate := builder.WithPredicates(predicate.Funcs{
 		CreateFunc: func(e event.CreateEvent) bool {
-			return isPrivateKeySecret(e.Object, r.watchNamespace)
+			return isPrivateKeySecret(e.Object, r.watchNamespace) || isTlsSecret(e.Object, r.watchNamespace)
 		},
 		DeleteFunc: func(e event.DeleteEvent) bool {
 			return isPrivateKeySecret(e.Object, r.watchNamespace)
```

Comment:
With no reconcile for deletion event, the deleted secret may still be in use on Windows nodes. Is that the desired behavior?